perun/logic/stats.py

```diff
@@ -341,3 +482,184 @@ def _modify_stats_file(stats_filepath, stats_ids, stats_contents, modify_functio
         for idx in range(min(len(stats_ids), len(stats_contents))):
             modify_function(stats_records, stats_ids[idx], stats_contents[idx])
         _save_stats_to(stats_handle, stats_records)
+
+
+def _get_version_candidates(minor_checksum, minor_date):
+    """ Obtains successor minor versions that have the same date as the given minor version.
+
+    :param str minor_checksum: the minor version checksum
+    :param str minor_date: the date of the minor version
+
+    :return list: list of successor versions as tuples: (checksum, date)
+    """
+    candidates = []
+    try:
+        # Start iterating the minor versions at the supplied version
+        from_iter = vcs.walk_minor_versions(minor_checksum)
+        # However, the first generator result is the version itself, skip it
+        next(from_iter)
+        successor = vcs.get_minor_version_info(next(from_iter).checksum)
+        while successor.date == minor_date:
+            candidates.append(successor.checksum)
+            successor = vcs.get_minor_version_info(next(from_iter).checksum)
+    except (exceptions.VersionControlSystemException, StopIteration):
+        # Some unexpected git corruption or we ran out of minor versions, end
+        pass
+    return candidates
+
+
+def _get_version_info(minor_version):
```

Comment:
This is only a helper function used to extract two members of the version object since the stats module works primarily with them and doesn't need the rest of the fields. I'm not sure if such function would be useful in the vcs package since it's a simple wrapper.